### PR TITLE
Emit warning on `chainerx` import in debug mode

### LINKED ISSUE #7177: Emit warning if compiled in Debug mode
In debug mode ChainerX runs significantly slower.
However sometimes it's difficult notice that.

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #7177

### CODE REVIEW

File: chainerx/__init__.py [hunk 1]
```diff
@@ -70,3 +71,9 @@ def __init__(self, *args, **kwargs):
 
 def is_available():
     return _available
+
+
+if _available and _core.is_debug():
```
Comment (asi1024) [inline chainerx/__init__.py]: How about emitting a warning if `chainer.is_debug() != chainerx.is_debug()`?

Comment (niboshi) [reply chainerx/__init__.py]: They are not intended to be aligned.
Perhaps it's better to change the name of `chainerx.is_debug`?

Comment (niboshi) [reply chainerx/__init__.py]: Made it private for now as discussed, as `chainerx._is_debug`.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
The Jenkins failure seems unrelated to this PR. LGTM.

**Pull request merged**